.github/workflows/build-test-distribute.yaml

```
@@ -170,3 +173,153 @@ jobs:
+          if [[ "$SKIP_CI_LABEL" == "true" ]]; then
+            echo "Skipping e2e jobs because the pull request contained a 'ci/skip-*' label."
+            exit 0
+          fi
+
+          JQ_PATH=$(which jq)
+          if [ "$JQ_PATH" == "" ]; then
+            echo "Could not find devtool 'jq', please check if 'check' job has run correctly prior to this job."
+            exit 1
+          fi
+
+          ALL_K8S_VERSION='{"k8sVersion":["kind", "kindIpv6", "${{ env.K8S_MIN_VERSION }}", "${{ env.K8S_MAX_VERSION }}"]}'
+          ALL_ARCH='{"arch": ["amd64", "arm64"]}'
+          ALL_TARGET='{"target": ["kubernetes", "universal", "multizone"]}'
+          SINGLE_ENV='{"k8sVersion": ["${{ env.K8S_MAX_VERSION }}"], "target": ["multizone"], "arch":["amd64"]}'
+          ALL_DELTA_KDS='{"deltaKDS": [true]}'
+          ALL_CNI='{"cniNetworkPlugin": ["calico"]}'
+
+          # GitHub Actions does not support ARM64
+          EXCLUDE_ARM64='[{"arch": "arm64"}]'
+
+          # kind should only be used when testing ipv6 or with e2e-universal
+          EXCLUDE_KIND='[{"k8sVersion":"kind", "target": "kubernetes"}, {"k8sVersion":"kind", "target": "multizone"}]'
+
+          # universal only runs on kind
+          EXCLUDE_UNIVERSAL='[{"target":"universal", "k8sVersion":"${{ env.K8S_MIN_VERSION }}"}, {"target":"universal", "k8sVersion":"${{ env.K8S_MAX_VERSION }}"}]'
+
+          # non priority main e2e
+          EXCLUDE_MAIN_NON_PRIORITY='[{"k8sVersion":"kindIpv6"}, {"k8sVersion": "${{ env.K8S_MIN_VERSION }}"}]'
+
+          # Default matrix
+          SLOW=$(echo '{}' | $JQ_PATH -rc ".exclude += $EXCLUDE_ARM64" | $JQ_PATH -rc ". += $ALL_K8S_VERSION | . += $ALL_ARCH")
+          MAIN=$(echo '{}' | $JQ_PATH -rc ".exclude += $EXCLUDE_ARM64" | $JQ_PATH -rc ". += $ALL_K8S_VERSION | . += $ALL_ARCH | . += $ALL_TARGET" | $JQ_PATH -rc ".exclude += $EXCLUDE_KIND | .exclude += $EXCLUDE_UNIVERSAL" )
+          DELTA_KDS=$(echo '{}' | $JQ_PATH -rc ".exclude += $EXCLUDE_ARM64" | $JQ_PATH -rc ". += $ALL_DELTA_KDS | . += $SINGLE_ENV")
+          CNI=$(echo '{}' | $JQ_PATH -rc ".exclude += $EXCLUDE_ARM64" | $JQ_PATH -rc ". += $ALL_CNI | . += $SINGLE_ENV")
+
+          RUN_FULL_MATRIX="${{ contains(github.event.pull_request.labels.*.name, 'ci/run-full-matrix') }}"
+          if [[ "$RUN_FULL_MATRIX" == "false" ]]; then
+            echo "Skipping non priority e2e tests, because no label 'ci/run-full-matrix' was found on the pull request."
+
+            SLOW=''         # target==""
+            DELTA_KDS=''    # deltaKDS=="true"
+            CNI=''          # cniNetworkPlugin=="calico"
+            MAIN=$(echo $MAIN | $JQ_PATH -rc ".exclude += $EXCLUDE_ARM64 | .exclude += $EXCLUDE_MAIN_NON_PRIORITY")
+          fi
+
+          echo "slow=$SLOW" >> $GITHUB_OUTPUT
+          echo "main=$MAIN" >> $GITHUB_OUTPUT
+          echo "delta_kds=$DELTA_KDS" >> $GITHUB_OUTPUT
+          echo "cni=$CNI" >> $GITHUB_OUTPUT
+  e2e_slow:
```

```suggestion
  test/e2e:
```

THis way way it's the same as the code folder

---

(don't know if `/` is valid in keys though)

---

Yes, just tried and the slash `/` is not allowed. Maybe it's because it will be used as a field name in some scenarios.

---

Maybe just test_e2e and test_e2e_env